src/lib/libast/tests/string/chresc.c

```diff
@@ -0,0 +1,117 @@
+#include "config_ast.h"  // IWYU pragma: keep
+
+#include <stdint.h>
+#include <string.h>
+
+#include "ast.h"
+#include "terror.h"
+
+struct test {
+    const char *input;
+    int result;
+};
+
+// Test for octal numbers. It also checks if next string is returned correctly.
+void test_octal() {
+    char *input = "\\012\\345\\670";
+    char *nextstr = input;
+    int next, expected;
+
+    next = chresc(input, &nextstr);
+
+    expected = 10;
```

Comment:
Consider replacing decimal constants with octal constants to make the relationship to the `input` string clearer.